routers/web/repo/pull.go

```diff
@@ -1313,6 +1313,37 @@ func MergePullRequest(ctx *context.Context) {
 			return
 		}
 
+		pullRequestsToHead, err := issues_model.GetUnmergedPullRequestsByBaseInfo(ctx, pr.HeadRepoID, pr.HeadBranch)
+		if err != nil {
+			ctx.ServerError("GetUnmergedPullRequestsByBaseInfo", err)
+			return
+		}
+
+		for _, prToHead := range pullRequestsToHead {
+			if prToHead.BaseRepoID != pr.BaseRepoID {
+				continue
+			}
+
+			if err := prToHead.LoadIssue(ctx); err != nil {
+				ctx.ServerError(fmt.Sprintf("LoadIssueForPullRequest[%d]", prToHead.ID), err)
+				return
+			}
+
+			if prToHead.Issue.RepoID == pr.Issue.RepoID {
+				prToHead.Issue.Repo = pr.Issue.Repo
+			} else {
+				if err := prToHead.Issue.LoadRepo(ctx); err != nil {
+					ctx.ServerError(fmt.Sprintf("LoadRepoForIssue[%d]", prToHead.IssueID), err)
+					return
+				}
+			}
+
+			if err := pull_service.ChangeTargetBranch(ctx, prToHead, ctx.Doer, pr.BaseBranch); err != nil {
+				ctx.ServerError(fmt.Sprintf("ChangeTargetBranch[%d]", prToHead.ID), err)
+				return
+			}
+		}
```

Comment:
Could we please extract this code into a new subfunction?
Something like `redirectOpenPulls` or so?

Reply:
I think it's a good idea. This will allow you to use redirection in the API, as @kvaster noted.